libvirt.go

```diff
@@ -142,6 +143,52 @@ const (
 	UndefineFlagNVRAM
 )
 
+// Capabilities returns an XML document describing the host's capabilties.
+func (l *Libvirt) Capabilities() ([]byte, error) {
+	resp, err := l.request(constants.ProcConnectGetCapabilties, constants.ProgramRemote, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	r := <-resp
+	if r.Status != StatusOK {
+		return nil, decodeError(r.Payload)
+	}
+
+	dec := xdr.NewDecoder(bytes.NewReader(r.Payload))
+	caps, _, err := dec.DecodeString()
+
+	return []byte(caps), err
+}
+
+// CPUFlags returns a list of CPU flags supported by the host.
+func (l *Libvirt) CPUFlags() ([]string, error) {
```

Comment:
It feels a little odd to have our own call that isn't part of the libvirt API.

Perhaps it would be beneficial to make a higher-level type (like `qemu.Domain`) and put these kinds of functions on that type, which wraps the raw libvirt type.